### PR TITLE
Rename pdf

### PR DESCRIPTION
**Pull request by david-hopper**
pyXain now renames the manuscript file to the contract hash before submitting to IPFS. This will make finding a paper on IPFS very easy with access to the blockchain. Upon finding a particular paper version's hash on the blockchain, one could acquire the pdf file and meta data by entering the following with an ipfs node running

```
http://localhost:8080/ipfs/<paper-hash>/meta.json
http://localhost:8080/ipfs/<paper-hash>/<contract-hash>.pdf
```

Ideally this would be `ipfs/paper-hash/v1/<contract-hash>.pdf` but there is a bug in the go-ipfs API preventing recursive adding of directories.

### CODE REVIEW

**Comment by david-hopper** [comment]
I used this branch to also fix some of the issues we had with installing IPFS in travisCI. TL;DR the path of the ipfs executable had to be appended, as opposed to pre-pended, to the system path. More tests were added as well to cover some more ground in pyXain and newly added functions for circumventing the faulty ipfs API for recursively adding.

File: node_client/pyXain/pyXain/pyXain.py [hunk 1]
```diff
@@ -208,19 +210,29 @@ def submit_manuscript(self, author_id, paper_id, paper_directory):
         author_check = self.check_author(author_id)
         if not author_check['Success'] :
             output = {}
-            output['Succcess'] = False
+            output['Success'] = False
```
Comment (SaulAryehKohn) [inline node_client/pyXain/pyXain/pyXain.py]: Noooo was this what was going wrong in that test? Too many c's?

File: node_client/pyXain/pyXain/pyXain.py [hunk 2]
```diff
@@ -384,7 +407,14 @@ def submit_comment(self, author_id, paper_id, comment_directory):
         # large directory. More recent versions have this, but I haven't tried building
         # them from source to try out. This should be remedied in the future.
         try :
-            reply = self.api.add(version_path, recursive=True)
+            # go-ipfs api v0.4.7 has a bug that prevents recursively adding directories.
```
Comment (SaulAryehKohn) [inline node_client/pyXain/pyXain/pyXain.py]: Why does this comment+functionality occur in two places? Can it be made into its own function?

File: node_client/pyXain/pyXain/pyXain.py (hunk 1 above)
Comment (david-hopper) [reply node_client/pyXain/pyXain/pyXain.py]: Yes it was, there was a key error that lead me to it. -_-

File: node_client/pyXain/pyXain/pyXain.py (hunk 2 above)
Comment (david-hopper) [reply node_client/pyXain/pyXain/pyXain.py]: There is a lot of code redundancy in the submit_manuscript, submit_revision, and submit_comment that I want to eliminate in a future PR. I could add that to this one though as it is probably something I'll work on next.

Comment (SaulAryehKohn) [reply node_client/pyXain/pyXain/pyXain.py]: OK

**Comment by SaulAryehKohn** [comment]
Hmm. Merging is blocked because at least one of your commits is not signed?

**Comment by david-hopper** [comment]
I was able to merge it in

**Pull request merged**